rapids_build_backend/impls.py

```diff
@@ -271,27 +159,41 @@ def _edit_pyproject(config):
     being built. This is useful for projects that want to build wheels
     with a different name than the package name.
     """
-    pyproject = _get_pyproject()
-    project_data = pyproject["project"]
-    project_data["name"] += _get_cuda_suffix(config.require_cuda)
-
-    dependencies = pyproject["project"].get("dependencies")
-    if dependencies is not None:
-        project_data["dependencies"] = _process_dependencies(
-            config, project_data["dependencies"]
-        )
-
-    optional_dependencies = pyproject["project"].get("optional-dependencies")
-    if optional_dependencies is not None:
-        project_data["optional-dependencies"] = {
-            extra: _process_dependencies(config, deps)
-            for extra, deps in optional_dependencies.items()
-        }
-
     pyproject_file = "pyproject.toml"
     bkp_pyproject_file = ".pyproject.toml.rapids-build-backend.bak"
+
+    cuda_version = _get_cuda_version(config.require_cuda)
+
+    with open(config.dependencies_file) as f:
+        parsed_config = yaml.load(f, Loader=yaml.FullLoader)
+    files = {}
+    for file_key, file_config in parsed_config["files"].items():
+        if "pyproject" not in get_requested_output_types(file_config["output"]):
+            continue
+        pyproject_dir = os.path.join(
+            os.path.dirname(config.dependencies_file),
+            file_config.get("pyproject_dir", default_pyproject_dir),
+        )
+        if not os.path.exists(pyproject_dir):
+            continue
+        if not os.path.samefile(pyproject_dir, "."):
+            continue
+        file_config["output"] = ["pyproject"]
+        if config.matrix:
+            file_config["matrix"] = generate_matrix(config.matrix)
+        if cuda_version is not None:
+            file_config.setdefault("matrix", {})["cuda"] = [
+                f"{cuda_version[0]}.{cuda_version[1]}"
+            ]
+        files[file_key] = file_config
+    parsed_config["files"] = files
```

Comment:
I'm not comfortable with the degree to which we're forced to reimplement dfg functionality here. We definitely need to think about how best to expose these features.